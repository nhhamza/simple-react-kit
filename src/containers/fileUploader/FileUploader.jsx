import React, { useState } from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import PropTypes from 'prop-types';
import { Button } from '@material-ui/core';
import Dropzone from '../../components/dropzone/Dropzone';
import t from '../../constants/translates';
import { selectedFiles } from '../../redux/file-uploader/fileUploader.selectors';
import { addFiles } from '../../redux/file-uploader/fileUploader.actions';

const FileUploader = ({ getSelectedFiles, addFiles }) => {
  const [files, setFiles] = useState([]);

  const addFilesHandler = () => {
    addFiles(files);
  };

  const handleChange = files => {
    setFiles({
      files,
    });
  };
  const handleSave = files => {
    setFiles({
      files,
    });
  };

  return (
    <div>
      <Dropzone
        onChange={handleChange}
        onSave={handleSave}
        dropzoneText={t?.fileUploader.textAreaDescription}
      />
      <br />

      <Button
        variant="contained"
        color="secondary"
        className="button button__up"
        type="button"
        onClick={addFilesHandler}
      >
        Suivant
      </Button>
    </div>
  );
};

FileUploader.propTypes = {
  addFiles: PropTypes.func,
  getSelectedFiles: PropTypes.arrayOf(PropTypes.oneOfType([PropTypes.string])),
};

const mapStateToProps = createStructuredSelector({
  getSelectedFiles: selectedFiles,
});

const mapDispatchToProps = dispatch => ({
  addFiles: files => dispatch(addFiles(files)),
});

export default connect(mapStateToProps, mapDispatchToProps)(FileUploader);
