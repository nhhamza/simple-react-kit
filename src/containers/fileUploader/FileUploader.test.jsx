import React from 'react';
import { mount } from 'enzyme';

import FileUploader from './FileUploader';

describe('<FileUploader />', () => {
  const wrapper = mount(<FileUploader />);

  it('renders', () => {
    expect(wrapper.exists()).toBeTruthy();
  });
});
