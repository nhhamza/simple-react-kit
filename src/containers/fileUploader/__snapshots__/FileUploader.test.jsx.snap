// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FileUploader /> renders 1`] = `true`;
